parts/windows/windowscsehelper.ps1

```diff
@@ -0,0 +1,47 @@
+# Define all exit codes in Windows CSE
+
+function Set-ExitCode
+{
+    Param(
+        [Parameter(Mandatory=$true)][int]
+        $ExitCode,
+        [Parameter(Mandatory=$true)][string]
+        $ErrorMessage
+    )
+    Write-Log "Set ExitCode to $ExitCode and exit. Error: $ErrorMessage"
+    $global:ExitCode=$ExitCode
+    $global:ErrorMessage=$ErrorMessage
+    exit $ExitCode
+}
+
+function Postpone-RestartComputer
+{
+    Write-Log "Creating an one-time task to restart the VM"
+    $action = New-ScheduledTaskAction -Execute "powershell.exe" -Argument " -Command `"Restart-Computer -Force`""
+    $principal = New-ScheduledTaskPrincipal -UserId SYSTEM -LogonType ServiceAccount -RunLevel Highest
+    # trigger this task once
+    $trigger = New-JobTrigger -At  (Get-Date).AddSeconds(15).DateTime -Once
+    $definition = New-ScheduledTask -Action $action -Principal $principal -Trigger $trigger -Description "Restart computer after provisioning the VM"
+    Register-ScheduledTask -TaskName "restart-computer" -InputObject $definition
+    Write-Log "Created an one-time task to restart the VM"
+}
+
+$global:WINDOWS_CSE_ERROR_UNKNOWN=1 # For unexpected error caught by the catch block in kuberneteswindowssetup.ps1

```

Comment:
I think we don't have to sync the exit error codes with Linux ones, do we have something on top of AKS node to analyze these output?
cc @xuto2

Reply:
Yes. We do not sync the exit error codes with Linux ones.
We will parse the exit code in AKS RP and return it to the customer.